internal/service/events/rule_data_source.go

```diff
@@ -0,0 +1,116 @@
+package events
+
+import (
+	"fmt"
+	"log"
+
+	"github.com/aws/aws-sdk-go-v2/aws"
+	"github.com/aws/aws-sdk-go/service/eventbridge"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema"
+	"github.com/hashicorp/terraform-provider-aws/internal/conns"
+	tftags "github.com/hashicorp/terraform-provider-aws/internal/tags"
+)
+
+func DataSourceRule() *schema.Resource {
+	return &schema.Resource{
+		Read: dataSourceRuleRead,
+
+		Schema: map[string]*schema.Schema{
+			"name": {
+				Type:     schema.TypeString,
+				Required: true,
+			},
+			"event_bus_name": {
+				Type:     schema.TypeString,
+				Optional: true,
+			},
+			"tags": tftags.TagsSchemaComputed(),
+			"arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"event_pattern": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"schedule_expression": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"state": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"description": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+			"created_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"managed_by": {
+				Type:     schema.TypeString,
+				Computed: true,
+			},
+			"role_arn": {
+				Type:     schema.TypeString,
+				Computed: true,
+				Optional: true,
+			},
+		},
+	}
+}
+
+func dataSourceRuleRead(d *schema.ResourceData, meta interface{}) error {
+	conn := meta.(*conns.AWSClient).EventsConn
+	ignoreTagsConfig := meta.(*conns.AWSClient).IgnoreTagsConfig
+
+	name := d.Get("name").(string)
+
+	input := &eventbridge.DescribeRuleInput{
+		Name: aws.String(name),
+	}
+
+	if v, ok := d.GetOk("event_bus_name"); ok {
+		input.EventBusName = aws.String(v.(string))
+	}
+
+	log.Printf("[DEBUG] Reading EventBridge rule (%s)", d.Id())
+	output, err := conn.DescribeRule(input)
+	if err != nil {
+		return fmt.Errorf("error getting EventBridge rule (%s): %w", d.Id(), err)
+	}
+
+	if output == nil {
+		return fmt.Errorf("error getting EventBridge rule (%s): empty response", d.Id())
+	}
+
+	log.Printf("[DEBUG] Found EventBridge rule: %#v", *output)
+
+	d.Set("arn", output.Arn)
+	d.Set("name", output.Name)
+	d.Set("state", output.State)
+	d.Set("created_by", output.CreatedBy)
+	d.Set("description", output.Description)
+	d.Set("managed_by", output.ManagedBy)
+	d.Set("role_arn", output.RoleArn)
+	d.Set("event_pattern", output.EventPattern)
+	d.Set("schedule_expression", output.ScheduleExpression)
+	d.Set("event_bus_name", output.EventBusName)
+
+	tags, err := ListTags(conn, *output.Arn)
+	if err != nil {
+		return fmt.Errorf("error listing tags for rule (%s): %w", name, err)
+	}
+	if err := d.Set("tags", tags.IgnoreAWS().IgnoreConfig(ignoreTagsConfig).Map()); err != nil {
+		return fmt.Errorf("error setting tags: %w", err)
+	}
+
+	d.SetId(*output.Arn)
```

Comment:
```suggestion
	d.SetId(aws.StringValue(output.Arn))
```

Reply:
That doesn't seem to be defined in `aws-sdk-go-v2/aws` ?

Reply:
Can you help @kamilturek ?

Reply:
Yes, that's correct. It's in `github.com/aws/aws-sdk-go/aws` which is used by all resources in the `events` package.

Reply:
So I should not use v2? The docs [state])https://hashicorp.github.io/terraform-provider-aws/aws-go-sdk-versions/):

> For new services, you should use [AWS SDK for Go v2](https://github.com/aws/aws-sdk-go-v2)

Reply:
@kamilturek I have updated the data source to use `aws-sdk-go`